src/persist/src/postgres.rs

```diff
@@ -197,14 +198,22 @@ impl PostgresConsensus {
     /// Open a Postgres [Consensus] instance with `config`, for the collection
     /// named `shard`.
     pub async fn open(config: PostgresConsensusConfig) -> Result<Self, ExternalError> {
+        // don't need to unredact here because we just want to pull out the username
+        let pg_config: Config = config.url.to_string().parse()?;
+        let role = pg_config.get_user().unwrap();
+        let create_schema = format!(
+            r#"CREATE SCHEMA IF NOT EXISTS "{}" AUTHORIZATION "{}""#,
+            "consensus", role
```

Comment:
Do we potentially need to do escaping on the role name? I'm not sure what is legal in a postgres role name.